server/src/main/java/org/elasticsearch/cluster/ClusterChangedEvent.java

```diff
@@ -92,6 +92,13 @@ public boolean indexRoutingTableChanged(String index) {
         return true;
     }
 
+    /**
+     * Returns <code>true</code> if the oldest index version changed in cluster metadata.
+     */
+    public boolean oldestIndexVersionChanged() {
+        return state.getMetadata().oldestIndexVersion() != previousState.getMetadata().oldestIndexVersion();
+    }
```

Comment:
ClusterChangeEvent has a lot of usages. I suggest to make it a `private static boolean oldestIndexVersionChanged(ClusterChangeEvent event)` next to its only usage to make the interface of the ClusterChangeEvent as small as possible. This could be revisited if we see this is needed in multiple places.

Reply:
Ok. I noticed ClusterChangeEvent has some of the utility methods, but code is years old.